File: math/0007-reverse-integer/reverse-integer-solution.cpp
class Solution {
public:
    int reverse(int x){
        while(x%10==0)
        {
            if((x<=9)&&(x>=-9))
                return x;
            else x/=10;
        }
        long long rev=0;
        int digit;
        do
        {
         digit = x % 10;
         rev = (rev * 10) + digit;
         x = x / 10;
        } while (x != 0);
        if((rev>INT_MAX)||(rev<INT_MIN))
            return 0;
        else return rev;
    }
};

File: math/0007-reverse-integer/reverse-integer-solution.java
class Solution {
    public int reverse(int x) {
        if (x == 0 || x / 10 == 0) return x;

        boolean isNegative = x < 0;
        x = Math.abs(x);

        StringBuilder res = new StringBuilder();

        while (true) {
            if (x / 10 == 0) {
                res.append(x);
                break;
            }
            res.append(x % 10);
            x /= 10;
        }

        try {
            int reversed = Integer.parseInt(res.toString());
            return isNegative ? -reversed : reversed;
        } 
        catch (NumberFormatException e) {
            return 0;
        }
    }
}
